packages/core/src/pagination/pagination.element.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2016-2021 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { baseStyles, LogService, querySlot } from '@cds/core/internal';
+import { html, LitElement } from 'lit-element';
+import { cache } from 'lit-html/directives/cache.js';
+import { CdsInput } from '../input/input.element.js';
+import { styles } from './pagination.element.css.js';
+
+export const PAGINATION_LAYOUT = 'horizontal gap:md align:center';
+
+/**
+ * Web component pagination.
+ *
+ * ```typescript
+ * import '@cds/core/pagination/register.js';
+ * ```
+ *
+ * ```html
+ * <cds-pagination>
+ *   <cds-pagination-button action="first" disabled></cds-pagination-button>
+ *   <cds-pagination-button action="prev" disabled></cds-pagination-button>
+ *   <span aria-label="current page">1 / 3</span>
+ *   <cds-pagination-button action="next"></cds-pagination-button>
+ *   <cds-pagination-button action="last"></cds-pagination-button>
+ * </cds-pagination>
+ * ```
+ *
+ * @element cds-pagination
+ * @slot
+ * @slot cds-pagination-button
+ * @cssprop --color
+ * @cssprop --font-size
+ */
+
+export class CdsPagination extends LitElement {
+  @querySlot('cds-input[cds-pagination-number]') private numberInput: CdsInput;
+
+  render() {
+    const userLayout = html`<slot></slot>`;
+    const defaultLayout = html`<div cds-layout="${PAGINATION_LAYOUT}"><slot></slot></div>`;
+
+    return html`${cache(this.hasAttribute('cds-layout') ? userLayout : defaultLayout)}`;
+  }
+
+  static get styles() {
+    return [baseStyles, styles];
+  }
+
+  updated(props: Map<string, any>) {
+    super.updated(props);
+    this.setupNumberInput();
+    this.validateAriaLabel();
+  }
+
+  private setupNumberInput() {
+    if (this.numberInput) {
+      this.numberInput.controlWidth = 'shrink';
+      this.numberInput.layout = 'compact';
+    }
+  }
+
+  private validateAriaLabel() {
+    if (!this.getAttribute('aria-label')?.length) {
```

Comment:
Similar to above we can add a prod check so dev time warnings dont run in prod env 

```typescript
if (notProductionEnvironment() && !this.getAttribute('aria-label')?.length) {
```